hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4632,14 +4633,47 @@ func (r *HostedClusterReconciler) reconcileKubevirtPlatformDefaultSettings(ctx c
 		}
 	}
 
+	// make sure there is an encryption key secret for etcd
+	if hc.Spec.SecretEncryption == nil {
+		etcdEnc, err := getOrCreateETCDEncKey(ctx, r.Client, hc)
+		if err != nil {
+			return err
+		}
+
+		hc.Spec.SecretEncryption = &hyperv1.SecretEncryptionSpec{
+			Type: hyperv1.AESCBC,
+			AESCBC: &hyperv1.AESCBCSpec{
+				ActiveKey: corev1.LocalObjectReference{
+					Name: etcdEnc.Name,
+				},
+			},
+		}
+	} else if hc.Spec.SecretEncryption.Type == hyperv1.AESCBC {
+		etcdEnc, err := getOrCreateETCDEncKey(ctx, r.Client, hc)
+
+		if err != nil {
+			return err
+		}
+
+		if hc.Spec.SecretEncryption.AESCBC != nil {
+			hc.Spec.SecretEncryption.AESCBC = &hyperv1.AESCBCSpec{}
+		}
+
+		if len(hc.Spec.SecretEncryption.AESCBC.ActiveKey.Name) == 0 {
+			hc.Spec.SecretEncryption.AESCBC.ActiveKey.Name = etcdEnc.Name
+		}
+	} else {
+		log.Error(fmt.Errorf("wrong encryption key type %s", hc.Spec.SecretEncryption.Type), "")
+	}
+
```

Comment:
```golang
       if hc.Spec.SecretEncryption == nil ||
                     hc.Spec.SecretEncryption.Type == "" ||
                     (hc.Spec.SecretEncryption.Type == hyperv1.AESCBC && hc.Spec.SecretEncryption.AESCBC == nil) {

               ### Maybe add this function to hypershift-operator/controllers/manifests/manifests.go to be consistent with how other objects are created for hosted cluster?
               secret := manifests.GeneratedAESCBCSecret(hc.Namespace, hc.Name) // <<<< this is in a new etcd package, to be also used from api/fixtures. No code in api/fixtures uses manifests

               ### use createOrUpdate function to handle the case where the secret was already created but we failed to apply the hostedCluster.spec change the first go around>
               if _, err := createOrUpdate(ctx, r.Client, secret, func() error {

                       ### don't override existing key just in case something weird happened >

                       _, exists := secret.Data[hyperv1.AESCBCKeySecretKey] // not sure about this code. This is just created and we know the key was generated
                       if exists {
                               return nil
                       }

                       ### If we generated the secret, then the secret needs to be cleaned up automatically when the HostedCluster is deleted>

                       ownerRef := config.OwnerRefFrom(hc)
                       ownerRef.ApplyTo(secret)

                       ### generate key and add to the secret's data
                 
                       return nil
               }); err != nil {
                       return fmt.Errorf("failed to generate ETCD SecretEncryption key for KubeVirt platform HostedCluster: %w", err)
               }

               ### apply new secret reference to hosted cluster
       }

```